pytest_inmanta/plugin.py

```diff
@@ -536,4 +537,16 @@ def check_serialization(self, resource: Resource) -> Resource:
 
     def clean(self) -> None:
         shutil.rmtree(os.path.join(self._test_project_dir, "libs", "unittest"))
+        self.finalize_all_handlers()
         self.create_module("unittest", initcf=get_module_data("init.cf"), initpy=get_module_data("init.py"))
+
+    def finalize_handler(self, handler: ResourceHandler) -> None:
+        versions = sorted(handler.cache.counterforVersion.keys())
+        for version in versions:
+            if handler.cache.is_open(version):
```

Comment:
```suggestion
            while handler.cache.is_open(version):
```

this has reference counting and may need to be closed multiple times.